### ISSUE TITLE
PSA: status of project & looking for maintainers!

### ISSUE THREAD

**Issue by mihkeleidast** [frctl/fractal#1167]
Hey everyone! First of all, a big thank you to everyone who has used or is still using Fractal - it is such a great community to be a part of.

**TLDR: Fractal is currently unmaintained, but looking for maintainers for the future.**

A couple of years ago me, @Chapabu and @LeBenLeBen took over the maintenance of this project. We wanted to keep Fractal afloat, as back then we were all using it as an everyday tool. However, as is often the case with open source projects, we _wanted_ to do more than we had time or other resources for. As such, most of our plans to add features or do big rewrites stalled. With time, as we've switched either work projects or jobs, we have lost the motivation to keep things moving here, as we aren't using Fractal every day anymore. I myself also embarked on some new missions a few months back, leaving behind all my projects that were using Fractal, so alas I have no resources or need to continue my work here.

I (and I believe Matt and Ben as well) would be willing to transfer ownership if someone would be willing to maintain Fractal going forward. I would consider Fractal pretty much feature-complete. I don't think there's much features to add, or that there is any point in starting major rewrites - those would probably live better as new standalone tools. But there is still some things left to do, e.g. responding to some support requests here or there or keeping the dependencies up do date, etc. Comment here or jump in our Discord to get that conversation started, if interested.

Thanks again and I hope to see y'all elsewhere on the interwebs!
Mihkel.

**Comment by Chapabu** [comment]
Thanks for posting this @mihkeleidast. There's nothing more I'd love than to continue working on Fractal, but it's become incredibly tricky to juggle..well...life. 

Unfortunately the pandemic took a lot out of me, and definitely made me re-evaluate some things in my life. One of those things was how much time I really want to spend working for free in the evenings when I could/should be looking after myself (both mentally and physically).

Do I still use Fractal? Yes. 

Would I be willing to work on Fractal if we were to enable sponsorship of some kind (either via Github or Open Collective)? Definitely - especially seeing as how the UK is currently run by a bunch of fools who have no idea of how the general public are affected by our current cost of living crisis.

Do I think it's fair to enable sponsorship for Fractal? Yes and no. I always think it's fair for FOSS maintainers to be compensated for time spent working on FOSS; but without a clear roadmap for Fractal that seems a little premature at this point.

**Comment by TheGameFreak720** [comment]
I'm looking for a tool to be able to display components using atomic design and this one seems to be the most up to date with lesser bloat in third party dependencies. I'm willing to contribute over time. I don't have tons of availability, but find this project needed as an alternative to storybook, that often has a lot of security vulnerabilities. If given some permissions, I can work on updating the dependencies at the very least. I can also take ownership and work on this project now or in the future. Mainly, I was looking for a way to display web components.

**Comment by LorenzoDossi** [comment]
Here at [basili.co](www.basili.co) we use it extensively and everyday for every project, I'd love to be able to keep this project up to date :)

**Comment by michtio** [comment]
@mihkeleidast after seeing this, and loving Fractal our team would be interested to take over development of Fractal.
Is there a direct way to communicate about this? Feel free to give me some information! Thanks

**Comment by gyuri-lajos** [comment]
I worked with Fractal 5 years ago. I really appreciated the approach Atomic design is the right way to go but need systems where through use the wholeness of the compositional approach can reveal the implicate emergent order. Much of the best ideas in this setup seem to live on in Lit.dev. Would be nice to see the system revived and be liberated from the gratuitous complitations that are but rearrangement of a broken wheel. If this comment resonates with anybody I would be glad to be of assistance. I would like to see a system where the fractal approach would revived to power what I like to call Lietate Software as Mutual Learning and a conversation in the Long tail of the Internet where our ability to articulate intents and systems gets the focus it deserves and liberate us from gratiutous complications, compromises that all stems from bad cosmologies and epistemologies that are promoted.

**Comment by wesleyboar** [comment]
That was difficult to parse, @gyuri-lajos, but I think I get it. [Thank you.]

[My perspective:] I found Fractal to have met my wants (flexibility), but not met the needs of my peers (who need it to be easy to use and grok and contribute to). With it I created a working system, but achieving it via Fractal was difficult and unintuitive, and newcomers to it find the same to be true (though much fault also lies with [my implementation](https://github.com/TACC/Core-Styles/tree/v2.25.3)).

I hope someone receives the reins of Fractal and maintains some of the flexibility that makes it great, while making it as easy to work with as it initially appears to be — maybe to do so is just adding to documentation 🤷 (I used an undocumented feature or two).

**Comment by sssduong** [comment]
So what's the latest update on this? @TheGameFreak720 and @michtio have expressed interest in helping to maintain this project. Is there any movement on this? Thx!

**Comment by thommyhh** [comment]
@mihkeleidast I'm still using Fractal and I know of a few other people, amongst others a web agency in Berlin, that are still using it. I really like to take over ownership and keep maintaining it. I already have some ideas for what to improve. One thing is the documentation. ;-)

How do we do proceed? Please contact me either here on GitHub or at thorben@webcoast.dk.

**Comment by mihkeleidast** [comment]
Hi everyone who has commented here and offered help - sorry for not being very active here, I've turned off most notifications from this org and rarely check in manually.

After a bit of discussion, here's how we see this going forwards:
- First of all, please aknowledge that most of the codebase is from circa 2016 and has not seen any major updates. To put it plainly, the code is old and hard to work with, and would need major overhauling to spark some joy. At the same time, there are not very many tests, so any sort of major overhaul is prone to break things.
- For potential contributors: any sort of first steps should be about filing issues about what sort of improvements are needed, how you'd go about making those changes (explaining rationale, etc). Then also filing those changes as PRs and helping review those submitted by others. We're not very keen to hand over to someone who has not contributed nothing at all. This may be a slow process, as I don't have too much free time on my hands & any changes would still require approval from me (or other old maintainers, if they can spare any time). If all goes well and we see good contributions over time, we can bump some contributors to maintainers.

There's also a question of whether it would be easier for someone who wants the reins to do a major overhaul to first work on a fork. If someone manages to modernize the codebase and provide a documented easy upgrade path to those who are still using the original, that's certainly another way to go about it.

Note that we don't have access to the Twitter / X account for the project (how much that channel matters nowadays is another question, but still worth mentioning), so promoting it there is hard.

We also don't have direct ownership of the domain, however whoever does seems to have kept it updated over the years :)

**Comment by adrienne** [comment]
Hey all,

@Schleuse seems to have a partially-rewritten fork with a LOT of work put into it; perhaps he'd like to become the new maintainer? https://github.com/Schleuse/fractal

**Comment by janbrasna** [comment]
There's actually a TODO breakdown of things, how things progressed in that fork:

https://github.com/Schleuse/fractal/blob/main/TODO.md

**Comment by de-es** [comment]
The more up-to-date version could be found here: https://github.com/sitepark/fractality
